OpenRA.Mods.Cnc/Traits/AttackPopupTurreted.cs

```diff
@@ -39,27 +39,33 @@ class AttackPopupTurretedInfo : AttackTurretedInfo, Requires<BuildingInfo>, Requ
 		public override object Create(ActorInitializer init) { return new AttackPopupTurreted(init, this); }
 	}
 
-	class AttackPopupTurreted : AttackTurreted, INotifyBuildComplete, INotifyIdle, IDamageModifier
+	class AttackPopupTurreted : AttackTurreted, INotifyBuildComplete, INotifyIdle, IDamageModifier, INotifyCreated
 	{
 		enum PopupState { Open, Rotating, Transitioning, Closed }
 
 		readonly AttackPopupTurretedInfo info;
 		readonly WithSpriteBody wsb;
-		readonly Turreted turret;
 
 		int idleTicks = 0;
 		PopupState state = PopupState.Open;
 		bool skippedMakeAnimation;
+		Turreted turret;
 
 		public AttackPopupTurreted(ActorInitializer init, AttackPopupTurretedInfo info)
 			: base(init.Self, info)
 		{
 			this.info = info;
-			turret = turrets.FirstOrDefault();
 			wsb = init.Self.Trait<WithSpriteBody>();
 			skippedMakeAnimation = init.Contains<SkipMakeAnimsInit>();
 		}
 
+		void INotifyCreated.Created(Actor self)
+		{
+			var armamentturrets = getArmaments().Select(x => x.Info.Turret).ToHashSet();
+
+			turret = self.TraitsImplementing<Turreted>().Where(x => armamentturrets.Contains(x.Name)).First();
```

Comment:
Won't you also be able to use `turrets` (from [`AttackTurreted#L25](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs#L25)) instead doing the trait lookup again?